src/outputs/histogram.cpp

```diff
@@ -472,12 +472,13 @@ void HistogramOutput::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm
     using namespace HDF5;
     H5P const pl_xfer = H5P::FromHIDCheck(H5Pcreate(H5P_DATASET_XFER));
 
-    // As we're reusing the interface from the existing hdf5 output, we have to define
-    // everything as 7D arrays.
-    // Counts will be set for each histogram individually below.
-    const std::array<hsize_t, H5_NDIM> local_offset({0, 0, 0, 0, 0, 0, 0});
-    std::array<hsize_t, H5_NDIM> local_count({0, 0, 0, 0, 0, 0, 0});
-    std::array<hsize_t, H5_NDIM> global_count({0, 0, 0, 0, 0, 0, 0});
+    // As we're reusing the interface from the existing hdf5 output,
+    // we have to define everything as H5_NDIM arrays. Counts will be
+    // set for each histogram individually below. All
+    // zero-initialized
+    const std::array<hsize_t, H5_NDIM> local_offset = {0};
+    std::array<hsize_t, H5_NDIM> local_count = {0};
+    std::array<hsize_t, H5_NDIM> global_count = {0};
```

Comment:
Technically these arrays are zero-initialized no matter what I set in the initializer list. However, initializer lists initialize all the elements up to the length of the list, so if you have an array of dimension that changes, an initializer list of fixed size can be misleading. I set the first element of the array to zero just for clarity.